pepper-apis/src/main/java/org/broadinstitute/ddp/service/UserDeleteService.java

```diff
@@ -0,0 +1,381 @@
+            checkBeforeDelete(handle, user);
+        }
+        LogInfo logInfo = deleteUserSteps(handle, user, fullDelete);
+        LOG.warn("User {} deletion is completed SUCCESSFULLY. "
+                + "guid:{}, hruid:{}, e-mail:{}, name:{} {}, dob:{}\n"
+                + "Who deleted: {}\n"
+                + "Comment: {}\n"
+                + "Deleted data:\n{}",
+                fullDelete ? "FULL" : "SIMPLE",
+                user.getGuid(),
+                user.getHruid(),
+                user.getEmail(),
+                user.getProfile() != null ? objToStr(user.getProfile().getFirstName()) : "",
+                user.getProfile() != null ? objToStr(user.getProfile().getLastName()) : "",
+                user.getProfile() != null ? objToStr(user.getProfile().getBirthDate()) : "",
+                objToStr(whoDeleted),
+                objToStr(comment),
+                logInfo.getInfo());
+    }
+
+    private void checkBeforeDelete(Handle handle, User user) {
+        if (hasGovernedUsers(handle, user.getGuid())) {
+            throw new DDPException(format(ERROR_PREFIX_USER_DELETE + "the user has governed users", user.getGuid()));
+        }
+
+        // check if user refers to revisions
+        long[] revisionIds = handle.attach(JdbiRevision.class).findByUserId(user.getId());
+        if (revisionIds.length > 0) {
+            throw new DDPException(format(ERROR_PREFIX_USER_DELETE + "the user has references to a revision history", user.getGuid()));
+        }
+    }
+
+    private LogInfo deleteUserSteps(Handle handle, User user, boolean fullDelete) throws IOException {
+        LogInfo logInfo = new LogInfo();
+        UserCollectedData userCollectedData = new UserCollectedData();
+
+        if (fullDelete) {
+            deleteKitRequests(handle, user, logInfo);
+            deleteUserStudyLegacyData(handle, user, logInfo);
+            deleteUserAnnouncement(handle, user, logInfo);
+            deleteActivityInstanceStatusByOperator(handle, user, logInfo);
+            deleteAnswersByOperator(handle, user, logInfo);
+        }
+
+        deleteUserProfile(handle, user, logInfo);
+        deleteMedicalProvider(handle, user, logInfo);
+        deleteEnrollmentStatuses(handle, user, logInfo);
+        deleteUserAddresses(handle, user, logInfo);
+        deleteAnswersAndActivityInstances(handle, user, logInfo);
+        deleteCountersAndEvents(handle, user, logInfo);
+        deleteGovernances(handle, user, logInfo, userCollectedData);
+        deleteAgeUpCandidates(handle, user, logInfo);
+        deleteUser(handle, user, logInfo);
+        deleteElasticSearchData(handle, user, logInfo, userCollectedData);
+
+        if (fullDelete) {
+            deleteAuth0User(handle, user, logInfo);
+        }
+
+        dataSyncRequest(handle, userCollectedData);
+
+        return logInfo;
+    }
+
+    private void deleteKitRequests(Handle handle, User user, LogInfo logInfo) {
+        logInfo.add("kit_request");
```

Comment:
Collecting the tables touched by deletion into a LogInfo object is an interesting idea, but I don't think it is particularly useful to the caller of the method. There is nothing that the caller can do with the information. I would rather write to the real log the operations as we do them and see how many rows of each kind have been removed. And if something goes wrong with a step of the deletion, then a developer with access to logs can see how far it got from the log. The LogInfo is not even going to be returned in case of failure.

Reply:
Replaced this logging approach to separate log calls before each deletion step.